src/datasource.ts

```diff
@@ -385,6 +386,25 @@ const recordSchema = (columnName: string, schema: any, result: AdxColumnSchema[]
     // https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/scalar-data-types/dynamic#dynamic-object-accessors
     const key = `${columnName}["${name}"]`;
 
+    if (Array.isArray(schema[name])) {
+      // If a field can have different types (e.g. long and double)
+      // we select the first, assuming they are interchangeable
+      const defaultCslType = schema[name][0];
+      if (schema[name].every((t: string) => toPropertyType(t) === QueryEditorPropertyType.Number)) {
+        // If all the types are numbers, the double takes precedence since it has more precission.
+        const cslType = schema[name].find((t: string) => t === 'double' || t === 'real') || defaultCslType;
+        result.push({ Name: key, CslType: cslType });
+      } else {
+        console.warn(`schema ${name} may contain different types, assuming ${defaultCslType}`);
```

Comment:
There is not a much better way of doing this. Since this is the `datasource` code, we cannot `dispatch` a warning like in components to show a popup warning. We could throw an error but that would halt the schema processing and we don't want that. We could also modify the function signature to optionally return an error but that would be difficult to handle by the caller and a bit of an overkill for this corner case. WDYT?

Reply:
Thank you for the explanation 😊 I agree it's not worth modifying the signature for an edge case. I also think we shouldn't halt the schema processing. Lets keep it as is and if anything needs to be changed in the future it can be revisited, great stuff!